### PR TITLE
Reimplement vaccination models as a dictionary of models.

### PR DESCRIPTION
**Pull request by tobiasraabe**
Make vaccination models similar to all other models.

### CODE REVIEW

**Comment by tobiasraabe** [comment]
@roecla Hi Klara, could you review this PR so we can close both branches, here and at sid-germany?

File: src/sid/initial_conditions.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def sample_initial_distribution_of_infections_and_immunity(
         virus_strains (Dict[str, Any]): A dictionary with the keys ``"names"`` and
             ``"factors"`` holding the different contagiousness factors of multiple
             viruses.
-        vaccination_model (Optional[Callable]): A function accepting ``states``,
+        vaccination_models (Optional[Callable]): A function accepting ``states``,
             ``params``, and a ``seed`` which returns boolean indicators for individuals
             who received a vaccination.
```
Comment (roecla) [inline src/sid/initial_conditions.py:199]: I think you forgot to update the docstring here.

File: src/sid/validation.py [hunk 2]
```diff
@@ -210,19 +210,37 @@ def validate_testing_models(
...
+
+
+def validate_return_is_series_or_ndarray(x, model_name, model_group, index):
     if isinstance(x, (pd.Series, np.ndarray)):
         return pd.Series(data=x, index=index)
     else:
-        raise ValueError(f"'{when}' must always return a pd.Series or a np.ndarray.")
+        raise ValueError(
+            f"The model '{model_name}' of '{model_group}' does not return a "
+            f"pandas.Series or a numpy.array, but {x}."
```
Comment (roecla) [inline src/sid/validation.py:233]:
```suggestion
            f"pandas.Series or a numpy.array, but {type(x)}."
```

**Review by roecla — APPROVED** [review]

File: src/sid/initial_conditions.py (hunk 1 above)
Comment (tobiasraabe) [reply src/sid/initial_conditions.py:199]: Done

**Pull request merged**